### DISCUSSION TITLE
Validation not triggered on draft save

### DISCUSSION THREAD

**Discussion by christoph-reps** [payloadcms/payload#2807] [Feature Requests & Ideas]
### Link to reproduction

https://github.com/christoph-reps/payload/tree/validate-length-with-draft

### To Reproduce

- Add text to the seeded post which is longer than 20 chars.
- Click save draft

### Describe the Bug

Savings drafts seems to not trigger validation. 

If you create a post and publish it validation will be triggered.

This is not the case when you save it as a draft.
I guess as a user i would expect both buttons to behave the same for validation. Or thought a bit further builtin validation for min / max character fields should trigger onChange or onBlur for text fields

### Payload Version

1.8.5

**Comment by jmikrut** [comment]
Hey @christoph-reps — this is actually by design. Because drafts can be saved in an unfinished state, saving a draft does not enforce validation. 

We have discussed potentially building a flag to _**enable**_ validation for drafts, therefore removing the ability to save an incomplete draft, but this would be a new feature that would be opt-in.

What do you think?  I'll convert this to a Feature Request discussion and we can keep the conversation going there!

**Reply by unonweb** [reply]
Hi jmikrut,
it would be great if this one is going to be configurable soon. In my concept of draft/published it prevents validations completely. That's because I use "Save Draft" in order to save the changed doc as json for my static site generator and "Publish" means "push it to the CDN when all changes are done and previewed".

I tried enabling validation on drafts by fiddling in `node_modules/payload/dist/admin/components/forms/Form.index.js`:

... but honestly I don't know what I'm doing and actually none of the changes that I tried in this file has ever had an effect on payload (I restarted of course).

Thank you for your work.

**Reply by abernh** [reply]
This was added via the collection config `drafts.validate = true` -- see https://payloadcms.com/docs/versions/drafts#options

**Comment by christoph-reps** [comment]
Hey, 

making it configurable would be a good solution.

I think it depends of once workflow. If you create big pages with lot of fields you might want to be noticed about limitations earlier than just before the publish since you don't know the limitations while you create the pages.  As the docs mentioned we could write a function for the [description](https://payloadcms.com/docs/fields/overview#description) to notify the user how much characters are left but  that would elaborate if you have to write it for all fields. So I think a builtin notification for min / max limitations would be nice.

**Comment by unonweb** [comment]
Can anybody point me to the spot where I can enable validation in drafts? 
I also tried changing the `disableValidation` function in `node_modules/schema-utils/dist/validate.js` but without success.

**Reply by unonweb** [reply]
Not required any more. I must admit that it makes totally sense to validate on publish.
